File: Protogen/loxim_protocol/cpp/protocol/packages_data/TimetzPackage.cpp
#include <string.h>
#include <stdlib.h>
#include "TimetzPackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

TimetzPackage::TimetzPackage(\
		TimePackage* a_time,\
		int8_t a_tz
		)
{
	time=a_time;
	tz=a_tz;
}

TimetzPackage::TimetzPackage()
{
	time=NULL;
	tz=0;
}

TimetzPackage::~TimetzPackage()
{
	if(time) delete time;
	time=NULL;
}

void TimetzPackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	time->serializeContent(writer);
	writer->writeInt8(tz);
}

bool TimetzPackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	time=new TimePackage();
	if (!time->deserializeContent(reader)) return false;
	tz=reader->readInt8();
	return true;
}

bool TimetzPackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	TimetzPackage* p=(TimetzPackage*)_package;
	if (!((!getTime() && ! p->getTime())||(getTime() && p->getTime() && (getTime()->equals(p->getTime()))))) return false;
	if (!(getTz()==p->getTz())) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages_data/TimetzPackage.java
package pl.edu.mimuw.loxim.protocol.packages_data;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class TimetzPackage extends Package {
		public final static long ID=(long)13;
	
    	/**
**/
    	private TimePackage time=null;
    	/**
**/
    	private byte tz=0;
    	
	public TimetzPackage()
	{
	}

		public TimetzPackage(
		    	TimePackage a_time,
	        	byte a_tz    	)
	{
        	time=a_time;
        	tz=a_tz;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			time=new TimePackage();
time.deserializeContent(reader);
					tz=reader.readInt8();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			time.serializeContent(writer);
		
					writer.writeInt8(tz);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		TimetzPackage p=(TimetzPackage)obj;
	    if (!(((time==p.getTime())||((time!=null)&&(p.getTime()!=null)&&((time).equals(p.getTime())))))) 
return false;
		if (!((tz==p.getTz()))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public TimePackage getTime(){return time;};
		
    	public void setTime(TimePackage a_time){ time=a_time;};
		
        	public byte getTz(){return tz;};
		
    	public void setTz(byte a_tz){ tz=a_tz;};
		
    
	 
}


